doc/user-guide/Push-notifications.md

```diff
@@ -0,0 +1,204 @@
+Firstly, prepare the following files structure:
+
+* priv/
+    * ssl/
+        * rest_cert.pem - The REST endpoint certificate
+        * rest_key.pem - private key for the REST endpoint certificate
+    * apns/
+        * prod_cert.pem - Production APNS app certificate
+        * prod_key.pem - Production APNS app certificate's private key
+        * dev_cert.pem - Development APNS app certificate
+        * dev_jey.pem - Development APNS app certificate's private key
+
+Assuming that your _FCM_ app token is `MY_FCM_SECRET_TOKEN` and you have the `priv` directory with all ceriticates in current directory, then you may start MongoosePush with the following command:
+
+```bash
+docker run -v `pwd`/priv:/opt/app/priv \
+  -e PUSH_FCM_APP_KEY="MY_FCM_SECRET_TOKEN" \
+  -e PUSH_HTTPS_CERTFILE="ssl/rest_cert.pem" \
+  -e PUSH_HTTPS_KEYFILE="ssl/rest_key.pem" \
+  -it --rm mongooseim/mongoose-push:latest
+```
+
+If you don't want to use either _APNS_ or _FCM_, you simply need to pass `PUSH_APNS_ENABLED=0` or `PUSH_FCM_ENABLED=0` respectively as additional env variables in your `docker run` command.
+For more advanced options and configuration please refer to _"Quick start / Configuring"_ in [MongoosePush][]'s [_README.md_][MongoosePushReadme].
+
+When your [MongoosePush][] docker is up and running, Push Notifications now can be used in your MongooseIM instance.
+
+## Using push notifications on client side
+
+The XMPP client application has very few things to do in order to receive push notifications.
+The whole process described in this section along with example notification flow we can see in the following diagram:
+ ![](push_notifications_example_message_flow.svg)
+
+### Registering with Push Service provider
+
+Firstly, client application has to get a device-specific token from Push Service Provider (FCM or APNS). This process is different, depending on the platform, so please consult your Push Service Provider's manual to learn about how to get this token.
+For example, [here](https://firebase.google.com/docs/cloud-messaging/android/client) you can learn about setting up _FCM_ on _Android_ platform and [here](https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/HandlingRemoteNotifications.html#/apple_ref/doc/uid/TP40008194-CH6-SW1) you can learn about setting up _APNS_ on _iOS_ platform.
+
+After this step, your application shall be able to receive _FCM_ or _APNS_ token - it will be required in the next step of this tutorial.
+
+### Setting up XMPP `pubsub` node
+
+First thing that the clinet has to set up on XMPP server is the `PubSub`'s node for handling push notifications.
+Assuming that `mypubsub.com` is a domain of MongooseIM server that has [mod_pubsub][] enabled with `push` node support, the client shall send the following stanza to the server:
+
+```xml
+<iq type='set'
+    from='me@mongooseim.com'
```

Comment:
I didn't see domain `mongooseim.com` anywhere else in this article. Should it be `mychat.com` instead?